components/openzwave-adapter/src/lib.rs

```diff
@@ -241,10 +258,12 @@ impl OpenzwaveAdapter {
             setter_map: IdMap::new(),
             watchers: Arc::new(Mutex::new(Watchers::new())),
             value_cache: Arc::new(Mutex::new(HashMap::new())),
+            controller_map: IdMap::new(),
+            pairing_map: IdMap::new(),
         });
 
+        try!(box_manager.add_adapter(adapter.clone()));
         adapter.spawn_notification_thread(rx, box_manager);
-        try!(box_manager.add_adapter(adapter));
```

Comment:
I realized we need to add the adapter before adding any other things. Even if it's really unlikely it would be possible with the previous code.